src/matlab.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 The MathWorks, Inc.
+
+import * as core from "@actions/core";
+import * as http from "@actions/http-client";
+import * as io from "@actions/io";
+import * as tc from "@actions/tool-cache";
+import * as fs from "fs";
+import properties from "./properties.json";
+import * as script from "./script";
+
+export interface Version {
+    release: string;
+    semantic: string;
+}
+
+interface MATLABVersionInfo {
+    latest: string;
+    semantic: {
+        [release: string]: string | undefined
+    }
+}
+
+export async function toolcacheLocation(version: Version): Promise<string> {
+    let toolpath: string = tc.find("MATLAB", version.semantic);
+    if (toolpath) {
+        core.info(`Found MATLAB ${version.release} in cache at ${toolpath}`);
+    } else {
+        fs.writeFileSync(".keep", "");
+        toolpath = await tc.cacheFile(".keep", ".keep", "MATLAB", version.semantic);
+        io.rmRF(".keep");
+    }
+    return toolpath
+}
+
+export async function setupBatch(platform: string) {
+    const batchInstallDir = script.defaultInstallRoot(platform, "matlab-batch")
+    await script
+        .downloadAndRunScript(platform, properties.matlabBatchInstallerUrl, [batchInstallDir])
+        .then(() => {
+            core.addPath(batchInstallDir);
+        });
+    return
+}
+
+export async function getVersion(release: string): Promise<Version> {
+    const client: http.HttpClient = new http.HttpClient();
+    const versionInfo = await client.getJson<MATLABVersionInfo>(properties.matlabReleaseInfoUrl);
+
+    if (!versionInfo.result) {
+        return Promise.reject(Error(`Could not retrieve version info. Contact ci@mathworks.com if this problem persists.`));
```

Comment:
@mw-hrastega

Reply:
Unable to retrieve the version information.

Questions:

-Version of what?
-You say: "Contact ... if this problem persists." When will the problem persist and when will it be resolved? Can the issue be resolved unexpectedly?

Reply:
Yeah, I wasn't sure about this one. This error can happen if there is an internet connection problem on the Github Actions Runner or if we change our deployed file to something invalid. So if the network is down for a couple seconds, it could potentially fail then fix itself.

Reply:
This one is also outdated. I was thinking `Unable to retrieve MATLAB release info. Contact continuous-integration@mathworks.com if this problem persists.`

Reply:
Suggesting some minor changes:

Unable to retrieve the MATLAB release information. Contact MathWorks at continuous-integration@mathworks.com if the problem persists.

Reply:
updated